src/Orleans/Serialization/TypeUtilities.cs

```diff
@@ -80,6 +80,11 @@ internal static bool IsOrleansShallowCopyable(this Type t)
             return false;
         }
 
+        private static bool IsTypeFieldsShallowCopyable(TypeInfo typeInfo)
+        {
+            return typeInfo.GetFields().All(f => !(f.FieldType.IsEquivalentTo(typeInfo)) && IsOrleansShallowCopyable(f.FieldType));
```

Comment:
this is not compatible with .NET core, since TypeInfo does not inherit from Type nor there is an extension method. We can use typeInfo.AsType() in that case